src/NServiceBus.Core/Encryption/ConfigureRijndaelEncryptionService.cs

```diff
@@ -169,12 +169,14 @@ IList<byte[]> expiredKeys
         public static void RegisterEncryptionService(this EndpointConfiguration config, Func<IEncryptionService> func)
         {
             Guard.AgainstNull(nameof(config), config);
-            config.Settings.Set("EncryptionServiceConstructor", func);
+
+            config.EnableFeature<Encryptor>();
```

Comment:
I'd say in general the feature itself can usually determine best whether to enable/disable itself and quite a few features use prerequisites for that.